### PR TITLE
Introduce and apply align width mechanism to handle alignment in tight width tables

### PR DESCRIPTION
**Pull request by chekaninmaxim**
https://github.com/area9innovation/innovation/pull/1422

### CODE REVIEW

**Comment by chekaninmaxim** [comment]
[Issue with centered texts in Tables](https://trello.com/c/zzTzwvO8/8205-issue-with-centered-texts-in-tables)

File: lib/tropic/tropic_paragraph.flow [hunk 1]
```diff
@@ -27,7 +31,11 @@ export {
 
 }
 
-TRenderParagraph(words : ref [TParaWord], s : [ParagraphStyle]) -> TDynamicParagraph {
+TRenderParagraph(
+	words : ref [TParaWord],
+	s : [ParagraphStyle],
+	alignWidthM : Maybe<DynamicBehaviour<double>>
```
Comment (ykuzmenko73) [inline lib/tropic/tropic_paragraph.flow]: rename it, it is alignWidthBM

File: lib/tropic/tropic_paragraph.flow [hunk 2]
```diff
@@ -71,27 +79,29 @@ TRenderParagraph(words : ref [TParaWord], s : [ParagraphStyle]) -> TDynamicParag
 		TDynamicGroup2(stackChanges, initialWords, TGroupCombiner())
 	));
 
-	updateFn = \wi -> {
+	updateFn = \wi, alWM -> {
```
Comment (ykuzmenko73) [inline lib/tropic/tropic_paragraph.flow]: rename alWM into alignWidthM

File: lib/tropic/tropic_paragraph.flow [hunk 3]
```diff
@@ -122,18 +132,22 @@ TRenderParagraph(words : ref [TParaWord], s : [ParagraphStyle]) -> TDynamicParag
...
 			dynArrayPush(stackChanges, TGroupDelete(idx));
 		});
-		if (reflow) updateFn(getValue(aw));
+		if (reflow) executeReflow();
 	}
 
 	tropic = TConstruct(
 		[
-			\-> subscribe(aw, updateFn),
+			eitherFn(
```
Comment (ykuzmenko73) [inline lib/tropic/tropic_paragraph.flow]: please check performance impact and try to implement throttle here

**Comment by chekaninmaxim** [comment]
@ykuzmenko73 I have checked on high amount of paragraphs and with throttling it works faster on fast typing. difference is about 0.5 sec, on fast typing of 10 letters

**Pull request merged**